From 9206746f5d83332de6268f0ebcb0f5805f04c0b9 Mon Sep 17 00:00:00 2001
From: Amos Kong <akong@redhat.com>
Date: Sat, 1 Nov 2014 05:07:49 -0400
Subject: [acpi] pm: Only set power states of devices that are power manageable

Message-id: <1414818469-16233-1-git-send-email-akong@redhat.com>
Patchwork-id: 98663
O-Subject: [RHEL-7.1 kernel PATCH] ACPI / PM: Only set power states of devices that are power manageable
Bugzilla: 1142683
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>

From: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Bugzilla: 1142683
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8186251
Test: https://bugzilla.redhat.com/show_bug.cgi?id=1142683#c6

Make acpi_device_set_power() check if the given device is power
manageable before checking if the given power state is valid for that
device.  Otherwise it will print that "Device does not support" that
power state into the kernel log, which may not make sense for some
power states (D0 and D3cold are supported by all devices by
definition).

Tested-by: Yinghai Lu <yinghai@kernel.org>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
(cherry picked from commit 2c7d132a589077b31493b3ea82ac83b1f72c93e1)

Signed-off-by: Amos Kong <akong@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/device_pm.c b/drivers/acpi/device_pm.c
index a6d04c6..586835d 100644
--- a/drivers/acpi/device_pm.c
+++ b/drivers/acpi/device_pm.c
@@ -155,7 +155,8 @@ int acpi_device_set_power(struct acpi_device *device, int state)
  int result = 0;
  bool cut_power = false;
 
- if (!device || (state < ACPI_STATE_D0) || (state > ACPI_STATE_D3_COLD))
+ if (!device || !device->flags.power_manageable
+     || (state < ACPI_STATE_D0) || (state > ACPI_STATE_D3_COLD))
   return -EINVAL;
 
  /* Make sure this is a valid target state */
-- 
1.7.1